BEAM-12770: Documented maven-to-gradle conversion for Dataflow

pcoet opened the pull request:
* Document Maven-to-Gradle conversion for DataflowRunner.
* Create TODOs for other runner conversion steps.
* Change Gradle guidance to show Kotlin instead of Groovy.
* Tighten up writing and remove unnecessary stuff (too much wordz).
* Make numbering of procedures consistent.
* Remove command prompts (`$`, `PS>`) from commands, to make them easier to copy.
* Normalize line length in markdown.
* Simplify Gradle execute task so that it reads from `--args` instead of `-D exec.args`
* Update sample text creation step to use text of "King Lear", which is the same as the default for DirectRunner.

I think this is a lot more streamlined, but let me know... Thanks!

@lukecwik 
@TheNeuralBit 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[![Build python source distribution and wheels](https://github.com/apache/beam/workflows/Build%20python%20source%20distribution%20and%20wheels/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[![Python tests](https://github.com/apache/beam/workflows/Python%20tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[![Java tests](https://github.com/apache/beam/workflows/Java%20Tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

aaltay commented:
@kileys - could you please review this change?

kileys commented on website/www/site/content/en/get-started/quickstart-java.md:
```diff
@@ -19,462 +19,376 @@ See the License for the specific language governing permissions and
+   {{< shell powerShell >}}
+cd .\word-count-beam
+   {{< /shell >}}
+   The directory contains a **pom.xml** and a **src** directory with example
```
When do we use bold vs the red text? Pom.xml in the next section uses `

pcoet replied:
We should use bold for filenames and the red text for code. There's a grey area for filenames in code... I'll fix the pom.xml in the next section. Thanks!

pcoet replied:
Fixed.

kileys commented on website/www/site/content/en/get-started/quickstart-java.md:
```diff
@@ -19,462 +19,376 @@ See the License for the specific language governing permissions and
+      {{< /highlight >}}
+   3. At the end of the build script, add the following conditional dependency:
+      {{< highlight >}}
+if (project.hasProperty("dataflow-runner")) {
```
This will be different for the other runners right? Should we add one of the runner select blocks here?

pcoet replied:
That's a good question. My thought was that we would add the other runner logic to this block, too, so that we have a single conversion process. For the user, it's just a single copy-paste. Otherwise we need a step for each runner, at which point we should not bother with the conditional logic (`if (project.hasProperty(...)`), because we're assuming that the user is only going to try one runner. But I'm inclined to follow the Maven model and put all the "profiles" here. Let me know if you disagree.

kileys commented on website/www/site/content/en/get-started/quickstart-java.md:
```diff
@@ -19,462 +19,376 @@ See the License for the specific language governing permissions and
-     -Dexec.args="--runner=FlinkRunner --flinkMaster=<flink master> --filesToStage=target/word-count-beam-bundled-0.1.jar \
-                  --inputFile=sample.txt --output=/tmp/counts" -Pflink-runner
-
-You can monitor the running job by visiting the Flink dashboard at http://<flink master>:8081
```
Can we save this as a comment somewhere? Or is it printed in the console somewhere?

pcoet replied:
I can't think of a way to do it -- I'm open to suggestions. The problem is that the syntax is copyable by a button in the docs. If we add a comment to the runnable command line, and the user copies it, I think it might cause problems. For example, if they try to use the up and down keys to go back to the last command executed. ...

kileys commented on website/www/site/content/en/get-started/quickstart-java.md:
```diff
@@ -19,462 +19,376 @@ See the License for the specific language governing permissions and
 {{< /runner >}}
-
 {{< runner dataflow >}}
-Make sure you complete the setup steps at /documentation/runners/dataflow/#setup
```
Same with this?

pcoet replied:
I added an extra comment emphasizing that you have to follow the setup steps for your runner. Hopefully that will do it...

kileys reviewed (commented):
I haven't had a chance to run the commands yet, but only had a few minor comments/questions. Thanks!

pcoet commented:
@kileys I made changes, in response to your feedback:

- Fixed the inconsistent filename formatting
- Reworked "Run a pipeline" to emphasize the runner setup steps. Any additional runner-specific instructions should go in the runner sections, instead of in the runnable commands. I think this is better now.
- Responded to your other comments inline.

Thanks!

David

kileys commented:
LGTM
